### PR TITLE
cavs: enable power gating of unused memory banks for cavs 2.x hw

### PR DESCRIPTION
**Pull request by lbetlej**
The change adds power gating of unused memory banks for cavs 2.0/2.5 hardware on top of previously enabled flow for cavs 1.8.

### CODE REVIEW

File: src/arch/xtensa/boot_loader.c [hunk 1]
```diff
@@ -168,9 +168,11 @@ static void parse_manifest(void)
 }
 #endif
 
-/* power off unused HPSRAM */
-#if CONFIG_CANNONLAKE
-
+#if CONFIG_CAVS_VERSION_1_8 || CONFIG_CAVS_VERSION_2_0 || \
```
Comment (abonislawski) [inline src/arch/xtensa/boot_loader.c]: Maybe "#if CAVS_VERSION >= CAVS_VERSION_1_8" is valid too?

File: src/arch/xtensa/boot_loader.c [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include <config.h>
 #include <stddef.h>
 #include <stdint.h>
+#include <cavs/version.h>
```
Comment (jajanusz) [inline src/arch/xtensa/boot_loader.c]: Includes are sorted by increasing visibility. `cavs/*` headers should be first in this file.

File: src/arch/xtensa/boot_loader.c (hunk 1 above)
Comment (lbetlej) [reply src/arch/xtensa/boot_loader.c]: ack

File: src/arch/xtensa/boot_loader.c (hunk 2 above)
Comment (lbetlej) [reply src/arch/xtensa/boot_loader.c]: ack

**Pull request merged**